client/state/user-settings/reducer.js

```diff
@@ -32,26 +33,30 @@ export const settings = ( state = {}, { type, settingValues } ) => {
 
 export const unsavedSettings = ( state = {}, action ) => {
 	switch ( action.type ) {
-		case USER_SETTINGS_UNSAVED_CLEAR:
-			// After a successful save, remove the saved settings (either all of them,
-			// or a subset) from the `unsavedSettings`.
-			if ( ! action.settingNames ) {
+		case USER_SETTINGS_UNSAVED_CLEAR: {
+			const settingNames = action.settingNames;
```

Comment:
We might as well have destructured this:

```suggestion
			const { settingNames } = action;
```